File: SW Expert Academy/D2/02005. 파스칼의 삼각형/파스칼의 삼각형.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
	int T; cin>>T;
	for(int tc=0; tc<T; tc++){
		int N; cin>>N;
		vector<vector<int>> nums;
		for(int i=0; i<N; i++){
			vector<int> v(N);
			nums.push_back(v);
		}
		
		printf("#%d\n", tc+1);
		
		nums[0][0] = 1;
		for(int i=1; i<N; i++){
			// 제일 왼쪽, 오른쪽은 1
			nums[i][0] = nums[i][i] = 1;
			for(int j=1; j<i; j++)
				nums[i][j] = nums[i-1][j-1]+nums[i-1][j];
		}
		
		for(int i=0; i<N; i++){
			for(int j=0; j<=i; j++)
				printf("%d ", nums[i][j]);
			cout<<endl;
		}
	}
}

File: SW Expert Academy/D2/02005. 파스칼의 삼각형/파스칼의 삼각형.py
for tc in range(int(input())):
    N = int(input())
    nums = [[0 for i in range(N)] for i in range(N)]

    print("#%d"%(tc+1))

    nums[0][0] = 1
    for i in range(1, N):
        nums[i][0] = nums[i][i] = 1
        for j in range(1, i):
            nums[i][j] = nums[i-1][j-1]+nums[i-1][j]

    for i in range(N):
        for j in range(i+1):
            print(nums[i][j], end=' ')
        print()
